src/libxtp/qmmolecule.cc

```diff
@@ -16,33 +16,58 @@
  *
  */
 
-#include <votca/csg/pdbwriter.h>
-#include <votca/csg/xyzreader.h>
-#include <votca/csg/xyzwriter.h>
+#include <votca/csg/io/pdbwriter.h>
+#include <votca/csg/io/xyzreader.h>
+#include <votca/csg/io/xyzwriter.h>
+#include <votca/csg/csgtopology.h>
 #include <votca/tools/elements.h>
 #include <votca/xtp/checkpointreader.h>
 #include <votca/xtp/checkpointwriter.h>
 #include <votca/xtp/qmmolecule.h>
+#include "../../include/votca/xtp/fileadapter.h"
 
 using namespace std;
 using namespace votca::tools;
 
 namespace votca {
 namespace xtp {
 
+const tools::DistanceUnit QMMolecule::distance_unit =                             
+    tools::DistanceUnit::angstroms;                                             
+const tools::MassUnit QMMolecule::mass_unit =                                     
+    tools::MassUnit::atomic_mass_units;                                         
+const tools::TimeUnit QMMolecule::time_unit = tools::TimeUnit::seconds;            
+const tools::ChargeUnit QMMolecule::charge_unit = tools::ChargeUnit::e;           
+const tools::EnergyUnit QMMolecule::energy_unit =                                 
+    tools::EnergyUnit::hartrees; 
+
+  tools::StructureParameters QMMolecule::getParameters() const {
+    tools::StructureParameters params;
+    params.set(tools::StructureParameter::AtomContainerType,getName());
+    params.set(tools::StructureParameter::AtomContainerId,getId());
+    return params;
+  }
+
 void QMMolecule::WriteXYZ(std::string filename, std::string header) const {
-  csg::XYZWriter writer;
+  csg::XYZWriter<csg::CSG_Topology> writer;
   writer.Open(filename, false);
-  writer.Write(*this, header);
+  writer.WriteHeader(header,size());
+  FileAdapter file_adapter;
```

Comment:
why is FileAdapter even a class it only has static functions?

Reply:
What would you suggest instead standalone functions?